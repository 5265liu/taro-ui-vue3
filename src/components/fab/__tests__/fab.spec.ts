import { mount } from '@vue/test-utils'
import AtFab from '../index'

const factory = (values = {}, slots = { default: ['按钮'] }) => {
  return mount(AtFab as any, {
    global: {
      components: {}
    },
    slots,
    props: { ...values },
  })
}

describe('AtFab Snap', () => {
  it('render AtFab -- default props', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtFab -- props class', () => {
    const wrapper = factory({ class: 'button' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtFab -- props size small', () => {
    const wrapper = factory({ size: 'small' })
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('AtFab Event', () => {
  it('AtFab onClick', () => {
    const onClick = jest.fn()
    const wrapper = factory({ onClick: onClick })

    wrapper.find('.at-fab').trigger('tap')
    expect(onClick).toBeCalled()
  })
})
