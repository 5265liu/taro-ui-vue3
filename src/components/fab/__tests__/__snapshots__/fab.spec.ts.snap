// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtFab Snap render AtFab -- default props 1`] = `
<view
  class="at-fab at-fab--normal"
>
  按钮
</view>
`;

exports[`AtFab Snap render AtFab -- props className 1`] = `
<view
  class="at-fab at-fab--normal"
>
  按钮
</view>
`;

exports[`AtFab Snap render AtFab -- props size small 1`] = `
<view
  class="at-fab at-fab--small"
>
  按钮
</view>
`;
